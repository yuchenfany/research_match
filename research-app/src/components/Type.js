/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import '../assets/index.css';
import { useNavigate } from 'react-router-dom';

function Type({ user, setUser }) {
  const navigate = useNavigate();

  function goToProfile() {
    if (user.type === 0) {
      navigate('/participant-profile');
    } else if (user.type === 1) {
      navigate('/researcher-profile');
    }
  }

  const updateType = async (value) => {
    let type = 0;
    if (value === 'researcher') {
      type = 1;
    }

    setUser({
      username: user.username,
      password: user.password,
      enrolled: user.enrolled,
      age: user.age,
      heightFeet: user.heightFeet,
      heightInches: user.heightInches,
      weight: user.weight,
      sex: user.sex,
      gender: user.gender,
      allergies: user.allergies,
      phys: user.phys,
      psych: user.psych,
      med: user.med,
      type,
    });
  };

  return (
    <div className="Type">
      <div className="header-left">Select account type</div>
      <label htmlFor="form" className="radio-option"> 
        <input
          type="radio"
          key="researcher"
          id="researcher"
          data-testid = "researcher"
          value="researcher"
          name="researcher"
          onClick={() => updateType('researcher')}
        />
        <div>Researcher</div>
      </label>
      <label htmlFor="form" className="radio-option">
        <input
          type="radio"
          key="participant"
          id="participant"
          data-testid = "participant"
          value="participant"
          name="participant"
          onClick={() => updateType('participant')}
        />
        <div>Participant</div>
      </label>
      <button className="confirm-button" type="submit" value="CONFIRM" onClick={goToProfile}> CONFIRM</button>
    </div>
  );
}

export default Type;
