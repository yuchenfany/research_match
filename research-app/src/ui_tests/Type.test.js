/**
 * @jest-environment jsdom
 */

/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-filename-extension */
import { render, screen, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
// import '@testing-library/jest-dom';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
// import { jest } from '@jest/globals';

import Type from '../components/Type';

// const mockedUsedNavigate = jest.fn();

// jest.mock('react-router-dom', () => ({
//   ...jest.requireActual('react-router-dom'),
//   useNavigate: () => mockedUsedNavigate,
// }));

test('Title label', () => {
    render(
        <Router>
            <Type />
        </Router>);  
  const label = screen.getByText('Select account type');
  expect(label).toBeInTheDocument();
});

test('participant clicked',() => {
  render(
    <Router>
        <Type />
    </Router>); 
// fireEvent.click(screen.getByTestId('participant'));
expect(screen.getByTestId('participant')).toBeInTheDocument();
//   expect(screen.getByTestId('participant')).toBeChecked();
});

test('Login Button',() => {
  const mockSetUser = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  render(
    <Router>
      <Type setUser={mockSetUser} user={user1} />
    </Router>,
  );
  const linkElement = screen.getByRole('button', {
    name: /confirm/i,
  });
  fireEvent.click(linkElement);
  // expect(mockSetUser).toBeCalled();
  expect(linkElement).toBeInTheDocument();
});

test('Researcher Button',() => {
  const mockSetUser = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  render(
    <Router>
      <Type setUser={mockSetUser} user={user1} />
    </Router>,
  );
  const linkElement = screen.getByRole('radio', {
    name: /researcher/i
  });
  fireEvent.click(linkElement);
  // expect(mockSetUser).toBeCalled();
  expect(linkElement).toBeInTheDocument();
});

test('Participant Button',() => {
  const mockSetUser = jest.fn();
  const user1 = { username: 'user1', password: 'user1pass' };
  render(
    <Router>
      <Type setUser={mockSetUser} user={user1} />
    </Router>,
  );
  const linkElement = screen.getByRole('radio', {
    name: /participant/i
  });
  fireEvent.click(linkElement);
  // expect(mockSetUser).toBeCalled();
  expect(linkElement).toBeInTheDocument();
});

test('researcher clicked',() => {
  render(
    <Router>
        <Type />
    </Router>); 
  // fireEvent.click(screen.getByTestId('researcher'));
  expect(screen.getByTestId('researcher')).toBeInTheDocument();
  // expect(screen.getByTestId('researcher')).toBeChecked();
});

test('snapshot test', () => {
  const component = renderer.create( 
    <Router>
        <Type />
    </Router>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});