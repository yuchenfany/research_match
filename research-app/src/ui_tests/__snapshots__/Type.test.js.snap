// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test 1`] = `
<div
  className="Type"
>
  <div
    className="header-left"
  >
    Select account type
  </div>
  <label
    className="radio-option"
    htmlFor="form"
  >
    <form>
      <label>
        researcher 
        <input
          data-testid="researcher"
          id="researcher"
          name="researcher"
          onClick={[Function]}
          type="radio"
          value="researcher"
        />
      </label>
    </form>
    <div>
      Researcher
    </div>
  </label>
  <label
    className="radio-option"
    htmlFor="form"
  >
    <form>
      <label>
        participant 
        <input
          data-testid="participant"
          id="participant"
          name="participant"
          onClick={[Function]}
          type="radio"
          value="participant"
        />
      </label>
    </form>
    <div>
      Participant
    </div>
  </label>
  <button
    className="confirm-button"
    onClick={[Function]}
    type="submit"
    value="CONFIRM"
  >
     CONFIRM
  </button>
</div>
`;
